main/src/library/Fixpoint3/Counter.flix

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright 2025 Casper Dalgaard Nielsen
+ *                Adam Yasser Tallouzi
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+///
+/// `Counter` is a wrapper around `Ref` for a simple counter.
+///
+/// Its primary use is for generating identifiers.
+///
+mod Fixpoint3 {
+    @Internal
+    pub enum Counter[r: Region] {
```

Comment:
Instead of using an enum, it would be more standard to use a struct. 

That also avoids the extra object.